feat(serverless): functions add note to runtime migration (#1602)
scaleway/docs-content: serverless/functions/reference-content/functions-lifecycle.mdx

| End Of Support (`end_of_support`) | It is no longer possible to create a new function with this runtime; however, functions that already use it can still be updated. A message with the date of End of Life is displayed. |
| End Of Life (`end_of_life`)       | Functions that use this runtime will still be working, but it is no more possible to update them.                                                                                      |

## Runtime upgrade

> To modify a runtime version you simply need to change the runtime value.

Reviewer: ```suggestion
To modify a runtime version, you simply need to change the runtime value.
```

> For example to migrate from `node18` to `node20` you need to change the runtime field and redeploy your function. Deployed runtimes are forward compatible but not necessary backward compatible.

Reviewer 2: ```suggestion
For example to migrate from `node18` to `node20`, change the runtime field and redeploy your function. Deployed runtimes are forward compatible but not necessarily backward compatible.
```

> Once a function is `deprecated`, `end_of_support` of `end_of_life` you are invited to upgrade your ongoing runtime to take advantage of new features, better performance and security fixes.

Reviewer: ```suggestion
Once a function is `deprecated`, `end_of_support` or `end_of_life`,  you are invited to upgrade your ongoing runtime to take advantage of new features, better performance and security fixes.
```
